From d52d96e8deee2f0489becc402b41db04abc8dc1c Mon Sep 17 00:00:00 2001
From: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
Date: Wed, 31 Jul 2024 11:20:48 +0200
Subject: [PATCH 8/8] onnxruntime: cmake: change visibility compilation options

Changing the compile options of the libonnxruntime shared lib
to export all the symbols of the lib and make them visible during
the runtime.

Signed-off-by: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
---
 cmake/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 2a18095d00..63cea7344d 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1803,6 +1803,8 @@ endif()

 message("ONNXRUNTIME_VERSION_MAJOR=${ONNXRUNTIME_VERSION_MAJOR}")
 set_target_properties(onnxruntime PROPERTIES SOVERSION "${ONNXRUNTIME_VERSION_MAJOR}")
+target_compile_options(onnxruntime PRIVATE -fvisibility=hidden)
+target_compile_options(onnxruntime PRIVATE -fvisibility=default)

 if(TARGET onnxruntime)
 # Install
--
2.34.1
